library/x509_crt.c

```diff
@@ -800,6 +800,21 @@ static int x509_get_crt_ext( unsigned char **p,
                 return( ret );
             break;
 
+        case MBEDTLS_X509_EXT_SUBJECT_KEY_IDENTIFIER:
+            if( ( ret = mbedtls_asn1_get_tag( p, end_ext_data, &len,
+                MBEDTLS_ASN1_OCTET_STRING ) ) == 0 )
```

Comment:
Minor: Please swap the branches here: return immediately if `ret != 0`, and otherwise proceed with the setup of `crt->subject_key_id`.